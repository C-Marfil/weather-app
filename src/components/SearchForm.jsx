import React from "react";
import PropTypes from "prop-types";
import "../styles/SearchForm.css";

const SearchForm = ({ searchText, setSearchText, onSubmit }) => {
  const handleInputChange = (event) => {
    setSearchText(event.target.value);
  };

  return (
    <div className="search-form">
      <input
        className="search-box"
        placeholder="Change Location"
        type="text"
        onChange={handleInputChange}
        value={searchText}
      />
      <button type="submit" onClick={onSubmit}>
        Location search
      </button>
    </div>
  );
};

SearchForm.propTypes = {
  searchText: PropTypes.string.isRequired,
  setSearchText: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
};

export default SearchForm;
