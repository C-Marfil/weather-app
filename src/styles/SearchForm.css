.search-form {
  display: flex;
  flex-direction: row;
  align-items: center;
  margin-bottom: 3%;
}

.search-box {
  background: #fff;
  height: 1vh;
  width: 8vw;
  border-radius: 30px;
  padding: 10px 5px;
  cursor: pointer;
}

.search-button {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 6px 14px;
  border-radius: 6px;
  border: none;
  background: #6e6d70;
  opacity: 0.8;
}

.search-button:focus {
  box-shadow: inset 0px 0.8px 0px -0.25px rgba(255, 255, 255, 0.2),
    0px 0.5px 1px rgba(0, 0, 0, 0.1), 0px 0px 0px 3.5px rgba(241, 242, 244, 0.5);
  outline: 0;
}
