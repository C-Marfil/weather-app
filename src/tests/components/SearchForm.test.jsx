import React from "react";
import { render } from "@testing-library/react";
import SearchForm from "../../components/SearchForm";

describe("Search Form feature", () => {
  const validProps = {
    searchText: "Liverpool",
    setSearchText: jest.fn(),
    onSubmit: jest.fn(),
  };

  it("Snapshot test", () => {
    const { asFragment } = render(<SearchForm />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("Renders the button correctly", () => {
    const { getByText } = render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.handleCitySearch}
      />
    );

    expect(getByText("Set Location")).toHaveAttribute("class", "search-button");
  });
});
