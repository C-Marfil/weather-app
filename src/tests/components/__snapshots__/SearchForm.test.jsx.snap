// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Form feature Snapshot test 1`] = `
<DocumentFragment>
  <div
    class="search-form"
  >
    <input
      class="search-box"
      placeholder="Location"
      type="text"
      value=""
    />
    <button
      class="search-button"
      type="submit"
    >
      Set Location
    </button>
  </div>
</DocumentFragment>
`;
